aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255.move

```diff
@@ -501,7 +508,7 @@ module aptos_std::ristretto255 {
 
     native fun scalar_invert_internal(bytes: vector<u8>): vector<u8>;
 
-    native fun scalar_from_sha512_internal(sha512_input: vector<u8>): vector<u8>;
+    native fun scalar_from_sha2_512_internal(sha2_512_input: vector<u8>): vector<u8>;
```

Comment:
This I can rename safely since not public, right @davidiw?

Reply:
You are right that this being a private function allows us to rename it. But still, this is a breaking change isn't it? I'm not sure if we can just check things like this in at this moment.

Reply:
Every change to native functions is complex. It is not a breaking change, but it need to be implemented in stages:

1. PR which makes changes to _native code only_. This can only introduce new native functions, not change existing ones.
2. Wait until this changes are governance approved and propagates sufficiently to all validators. For a time, some validators will have the new code but not yet execute it (and the old native functions are still being called).
3. PR which makes changes to Move code.
4. Framework upgrade via governance.

So _very_ complex. Please *absolutely be careful* to not land anything violating this (e.g. putting (1) and (4) into one PR and land this), because then you screw things up for everyone. Right now we do not have tools helping you here...

@movekevin @davidiw

Reply:
https://github.com/aptos-labs/aptos-core/pull/3755 has landed. I feel like this will cause the same issue one of @davidiw 's PR ran into, won't it?